numpy/_core/tests/test_strings.py

```diff
@@ -490,3 +490,178 @@ def test_strip_args(self, in1, in2, out, dt):
+        ("This is a tissue", "is", "**", MAX, "Th** ** a t**sue"),
+        ("This is a tissue", "is", "**", -1, "Th** ** a t**sue"),
+        ("This is a tissue", "is", "**", 4, "Th** ** a t**sue"),
+        ("This is a tissue", "is", "**", 3, "Th** ** a t**sue"),
+        ("This is a tissue", "is", "**", 2, "Th** ** a tissue"),
+        ("This is a tissue", "is", "**", 1, "Th** is a tissue"),
+        ("This is a tissue", "is", "**", 0, "This is a tissue"),
+        ("bobob", "bob", "cob", MAX, "cobob"),
+        ("bobobXbobobob", "bob", "cob", MAX, "cobobXcobocob"),
+        ("bobob", "bot", "bot", MAX, "bobob"),
+        ("Reykjavik", "k", "KK", MAX, "ReyKKjaviKK"),
+        ("Reykjavik", "k", "KK", -1, "ReyKKjaviKK"),
+        ("Reykjavik", "k", "KK", 2, "ReyKKjaviKK"),
+        ("Reykjavik", "k", "KK", 1, "ReyKKjavik"),
+        ("Reykjavik", "k", "KK", 0, "Reykjavik"),
+        ("A.B.C.", ".", "----", MAX, "A----B----C----"),
+        ("Reykjavik", "q", "KK", MAX, "Reykjavik"),
+        ("spam, spam, eggs and spam", "spam", "ham", MAX,
+            "ham, ham, eggs and ham"),
+        ("spam, spam, eggs and spam", "spam", "ham", -1,
+            "ham, ham, eggs and ham"),
+        ("spam, spam, eggs and spam", "spam", "ham", 4,
+            "ham, ham, eggs and ham"),
+        ("spam, spam, eggs and spam", "spam", "ham", 3,
+            "ham, ham, eggs and ham"),
+        ("spam, spam, eggs and spam", "spam", "ham", 2,
+            "ham, ham, eggs and spam"),
+        ("spam, spam, eggs and spam", "spam", "ham", 1,
+            "ham, spam, eggs and spam"),
+        ("spam, spam, eggs and spam", "spam", "ham", 0,
+            "spam, spam, eggs and spam"),
+        ("bobobob", "bobob", "bob", MAX, "bobob"),
+        ("bobobobXbobobob", "bobob", "bob", MAX, "bobobXbobob"),
+        ("BOBOBOB", "bob", "bobby", MAX, "BOBOBOB"),
+        ("one!two!three!", "!", "@", 1, "one@two!three!"),
+        ("one!two!three!", "!", "", MAX, "onetwothree"),
+        ("one!two!three!", "!", "@", 2, "one@two@three!"),
+        ("one!two!three!", "!", "@", 3, "one@two@three@"),
+        ("one!two!three!", "!", "@", 4, "one@two@three@"),
+        ("one!two!three!", "!", "@", 0, "one!two!three!"),
+        ("one!two!three!", "!", "@", MAX, "one@two@three@"),
+        ("one!two!three!", "x", "@", MAX, "one!two!three!"),
+        ("one!two!three!", "x", "@", 2, "one!two!three!"),
+        ("abc", "", "-", MAX, "-a-b-c-"),
+        ("abc", "", "-", 3, "-a-b-c"),
+        ("abc", "", "-", 0, "abc"),
+        ("abc", "ab", "--", 0, "abc"),
+        ("abc", "xy", "--", MAX, "abc"),
+    ])
+    def test_replace(self, buf, old, new, count, res, dt):
+        buf = np.array(buf, dtype=dt)
+        old = np.array(old, dtype=dt)
+        new = np.array(new, dtype=dt)
+        res = np.array(res, dtype=dt)
+        assert_array_equal(np.strings.replace(buf, old, new, count), res)
+
+    def test_replace_count_and_size(self, dt):
+        if dt != "U":
```

Comment:
Very minor detail, but I'd tend to just make this a separate test class if we're not using `dt` (which i agree makes sense).